### PR TITLE
Update Jackson BOM to 2.13.2.20220328

### PR DESCRIPTION
**Pull request by jorsol**
This fix a potential security issue https://github.com/advisories/GHSA-57j2-w4cx-62h2 in Jackson: https://github.com/FasterXML/jackson-databind/issues/2816

### CODE REVIEW

File: independent-projects/tools/pom.xml [hunk 1]
```diff
@@ -149,16 +149,6 @@
...
-            <dependency>
-                <groupId>com.fasterxml.jackson.core</groupId>
-                <artifactId>jackson-databind</artifactId>
-                <version>${jackson.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>com.fasterxml.jackson.dataformat</groupId>
-                <artifactId>jackson-dataformat-yaml</artifactId>
-                <version>${jackson.version}</version>
-            </dependency>
```
Comment (gsmet) [inline independent-projects/tools/pom.xml:161]: Given how CI reacted, I don't think this is right. You should import the BOM instead and get rid of the versions.

**Review by gsmet — DISMISSED** [review]

File: independent-projects/tools/pom.xml (hunk 1 above)
Comment (gsmet) [reply independent-projects/tools/pom.xml:161]: Ah wait, I just saw the BOM was at the bottom, we usually define it at the top.

**Comment by famod** [comment]
Gradle seems to have trouble getting the fixed databind dependency. Others seem to have similar issues: https://github.com/FasterXML/jackson-databind/issues/2816#issuecomment-1079512532

File: independent-projects/tools/pom.xml (hunk 1 above)
Comment (jorsol) [reply independent-projects/tools/pom.xml:161]: Yes, I was going to use the BOM but also saw it was a the bottom (which is odd). I can move it to the top.

Comment (gsmet) [reply independent-projects/tools/pom.xml:161]: @jorsol would you mind pushing this cleanup in a separate PR? Just the removal of these two artifacts and moving the BOMs at the top.
The situation with Jackson looks extremely weird on the Gradle side and I'm not sure we will be able to push this very update.

Comment (jorsol) [reply independent-projects/tools/pom.xml:159]: Yes, I could, but I honestly don't think that the cleanup is the problem.

Comment (gsmet) [reply independent-projects/tools/pom.xml:159]: I totally agree with you. The problem is on the Jackson side AFAICS. Thus why I would like to have the cleanup in another PR so we can merge it right away.
As for the update, I don't know. I had the same issue in another Gradle based project so I'm pretty sure there's something wrong with this new version.

**Comment by jorsol** [comment]
The current gradle issue is because the BOM of Jackson is not always 1:1 the same version of the components, for patch releases the Jackson BOM uses a fourth part that uses a date (here `.20220324`), this BOM only updates the databind component to `2.13.2.1` (with the `.1` patch release), but all other components remain at `2.13.2`, so we have:
`jackson-bom-2.13.2.20220324` -> `jackson-databind-2.13.2.1` -> `jackson-core-2.13.2`.
All this version mismatch confuse the resolver of gradle.

Gradle tries to resolve the bom version based on the version of the databind as seen here:

```
 Cause 1: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find com.fasterxml.jackson:jackson-bom:2.13.2.1.
Searched in the following locations:
  - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jackson-bom/2.13.2.1/jackson-bom-2.13.2.1.pom
  - file:/home/runner/.m2/repository/com/fasterxml/jackson/jackson-bom/2.13.2.1/jackson-bom-2.13.2.1.pom
Required by:
    project : > io.quarkus:quarkus-bom:999-SNAPSHOT > com.fasterxml.jackson.core:jackson-databind:2.13.2.1
Cause 2: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find com.fasterxml.jackson:jackson-bom:2.13.2.1.
Searched in the following locations:
  - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jackson-bom/2.13.2.1/jackson-bom-2.13.2.1.pom
  - file:/home/runner/.m2/repository/com/fasterxml/jackson/jackson-bom/2.13.2.1/jackson-bom-2.13.2.1.pom
Required by:
    project : > io.quarkus:quarkus-bom:999-SNAPSHOT > com.fasterxml.jackson.core:jackson-annotations:2.13.2
    project : > io.quarkus:quarkus-bom:999-SNAPSHOT > com.fasterxml.jackson.core:jackson-core:2.13.2
```

I'm not familiar with Gradle or know why is this happening, but if Quarkus "generates" the gradle sources, then is should be using an import like:
`implementation platform('com.fasterxml.jackson:jackson-bom:2.13.2.20220324')`

I try to rename the property to `jackson-bom.version` to better reflect what version should be there, just in case the property was generating a conflict with the imports, but it looks like it doesn't do any real effect (at least for Gradle).

So, I keep looking what might be causing this issue but a maintainer of the gradle bootstrap resolver could be of help here.

**Comment by jorsol** [comment]
Ok, yes, there is an issue with Jackson and Gradle: https://github.com/FasterXML/jackson-databind/issues/3428

**Comment by jorsol** [comment]
@gsmet a new jackson-bom with the fix for the Gradle Module Metadata is released, now the Gradle tests pass.

I leave the cleanup here since it doesn't break anything and I have renamed the property to `jackson-bom.version` since the bom version not always match the components version (and the bom should be used anyway), it would be more clear that the property refers to the bom version since components with a patch version don't necessarily match.

**Review by gsmet — APPROVED** [review]
Looks good, thanks. Let's see what CI has to say!

**Comment by jorsol** [comment]
Ready to go!

**Pull request merged**